boot/bootutil/src/image_validate.c

```diff
@@ -87,6 +87,29 @@ bootutil_img_hash(struct image_header *hdr, const struct flash_area *fap,
     return 0;
 }
 
+#if defined(MCUBOOT_SIGN_RSA) || defined(MCUBOOT_SIGN_EC) || \
+    defined(MCUBOOT_SIGN_EC256)
+static int
+bootutil_find_key(uint8_t *keyhash, int8_t keyhash_len)
+{
+    bootutil_sha256_context sha256_ctx;
+    int i;
+    const struct bootutil_key *key;
+    uint8_t hash[32];
+
+    for (i = 0; i < bootutil_key_cnt; i++) {
+        key = &bootutil_keys[i];
+        bootutil_sha256_init(&sha256_ctx);
```

Comment:
I wonder if we should come up with a way of computing these hashes at build time to compare that way?

Reply:
I was more concerned about the time, but SHA256 is pretty fast, even on these CPUs.